crates/matrix-sdk/src/widget/machine/mod.rs

```diff
@@ -203,6 +208,21 @@ impl WidgetMachine {
             FromWidgetRequest::SendEvent(req) => {
                 self.process_send_event_request(req, raw_request);
             }
+
+            FromWidgetRequest::GetOpenId {} => {
+                self.send_from_widget_response(raw_request, OpenIdResponse::Pending);
+                self.send_matrix_driver_request(RequestOpenId).then(|res, machine| {
+                    let response = match res {
+                        Ok(res) => OpenIdResponse::Allowed(OpenIdState::new(request_id, res)),
+                        Err(msg) => {
+                            info!("OpenID request failed: {msg}");
+                            OpenIdResponse::Blocked { original_request_id: request_id }
```

Comment:
Can you add a test where the OpenID request failed?

Reply:
I'll note it down for follow-up.

Reply:
Hm, there actually **was** a test for this in the former PR https://github.com/matrix-org/matrix-rust-sdk/pull/2743(`fn openid_fail_results_in_response_blocked()`). Will re-add it then 🙂